### PR TITLE
Scene settings save button updates

### PR DESCRIPTION
**Pull request by AMZN-stankowi**
## What does this PR do?

- Renamed "Update" button to "Save" button, to make it clear what the button does.
- The Save button is now always available, instead of just when the tool thinks the file is dirty.
- The timestamp of the scene manifest is now shown next to the Save button.
- The Save button gets an asterisk when the scene manifest is dirty, to call attention to it.
- Fixed the dirty check to handle toggles in manifest vector widgets. This logic now matches how the other interface elements are handled. This fixes the problem where toggling the "Ignore Y-Axis Transition" toggle wouldn't mark the scene manifest as dirty.
- The Save button uses the blue primary button styling.

image

image

image

## How was this PR tested?

Manual testing + new automated test

### CODE REVIEW

**Comment by rainbj** [comment]
So the only issue I have is showing the dirty flag on the save button. We should show it on the title header.
Example:

FBX Settings - Greencube*

This is how we are doing it in the main editor, Animation editor, and ScriptCanvas editor

File: Code/Editor/Plugins/EditorAssetImporter/ImporterRootDisplay.h [hunk 1]
```diff
@@ -40,25 +42,57 @@ namespace AZ
 
         }
     }
-}
+} // namespace AZ
```
Comment (jackalbe) [inline Code/Editor/Plugins/EditorAssetImporter/ImporterRootDisplay.h:45]: are these namespace end comments required?

File: Code/Editor/Plugins/EditorAssetImporter/ImporterRootDisplay.h [hunk 2]
```diff
@@ -40,25 +42,57 @@ namespace AZ
...
 
 namespace Ui
 {
     class ImporterRootDisplay;
-}
+} // namespace UI
 
-class ImporterRootDisplay 
+//! Python interface for the scene importer root display settings
+class SceneSettingsRootDisplayPythonRequests
```
Comment (jackalbe) [inline Code/Editor/Plugins/EditorAssetImporter/ImporterRootDisplay.h:53]: normally we do not specify a bus as a Python only bus, maybe a "Script" bus or just SceneSettingsRootDisplayRequests?

Note that C++ and Lua can also call this bus.

File: Code/Tools/SceneAPI/SceneUI/RowWidgets/ManifestVectorWidget.cpp [hunk 3]
```diff
@@ -271,8 +272,10 @@ namespace AZ
                                     return object.get() == cast;
                                 }) != m_manifestVector.end(), "ManifestVectorWidget detected an update of a field it doesn't own.");
                             EmitObjectChanged(cast);
+                            return;
```
Comment (jackalbe) [inline Code/Tools/SceneAPI/SceneUI/RowWidgets/ManifestVectorWidget.cpp:275]: is the return a good early out? Is one EmitObjectChanged() all that is needed?

**Review by amzn-rhhong — APPROVED** [review]
Great job on adding those tests too.

File: Code/Editor/Plugins/EditorAssetImporter/ImporterRootDisplay.h (hunk 1 above)
Comment (AMZN-stankowi) [reply Code/Editor/Plugins/EditorAssetImporter/ImporterRootDisplay.h:45]: no, but I've found them useful

File: Code/Tools/SceneAPI/SceneUI/RowWidgets/ManifestVectorWidget.cpp (hunk 3 above)
Comment (AMZN-stankowi) [reply Code/Tools/SceneAPI/SceneUI/RowWidgets/ManifestVectorWidget.cpp:275]: yeah, it's just verifying that it cares about the object change. This matches the logic in ManifestWidgetPage

File: Code/Editor/Plugins/EditorAssetImporter/ImporterRootDisplay.h (hunk 1 above)
Comment (hultonha) [reply Code/Editor/Plugins/EditorAssetImporter/ImporterRootDisplay.h:45]: We do call this out in the coding standards - https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#namespace-names (and `clang-format` will helpfully add them too)

**Review by hultonha — APPROVED** [review]
Change looks good to me and thanks for adding all the screen shots. I'd love to see the test output too if possible, otherwise looks great thanks!

I agree with @rainbj's suggestion about the Save button dirty flag too if that's not too hard to change.

File: Code/Editor/Plugins/EditorAssetImporter/ImporterRootDisplay.h (hunk 2 above)
Comment (AMZN-stankowi) [reply Code/Editor/Plugins/EditorAssetImporter/ImporterRootDisplay.h:53]: Good point, I've renamed this

**Comment by AMZN-stankowi** [comment]
> So the only issue I have is showing the dirty flag on the save button. We should show it on the title header. Example:
> 
> FBX Settings - Greencube*
> 
> This is how we are doing it in the main editor, Animation editor, and ScriptCanvas editor

We talked directly about this. Appending the star doesn't work because the title is elided, so you won't see it on long names. I've pre-pended the star instead.

**Comment by AMZN-stankowi** [comment]
> Change looks good to me and thanks for adding all the screen shots. I'd love to see the test output too if possible, otherwise looks great thanks!
> 
> I agree with @rainbj's suggestion about the Save button dirty flag too if that's not too hard to change.

Here is the test output. Ignore that the tests fail, I forced a fake failure to get the logs to print.

[scene_settings_tests.log](https://github.com/o3de/o3de/files/9725906/scene_settings_tests.log)

**Review by hultonha — APPROVED** [review]
Thanks for making the updates @AMZN-stankowi, LGTM :+1:

File: AutomatedTesting/Gem/PythonTests/assetpipeline/scene_settings_tests/tests/scene_settings_manifest_vector_widget_tests_in_editor.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            scene_test_helpers.prepare_scene_ui_for_test( \
+                test_file_name="Jack_Death_Fall_Back_ZUp.fbx", \
+                manifest_should_exist=True)
+        
+        # Find a toggle that used to not properly mark the UI as dirty because it was parented to a manifest vector widget.
+        y_axis = reflected_property_root.findChild(QtWidgets.QWidget,"Ignore Y-Axis transition")
+        Report.critical_result(tm.Test_Messages.scene_settings_y_axis_row_found, y_axis is not None)
+
+        check_boxes = y_axis.findChildren(QtWidgets.QCheckBox,"")
+        Report.critical_result(tm.Test_Messages.scene_settings_y_axis_check_box_found, check_boxes is not None)
```
Comment (jackalbe) [inline AutomatedTesting/Gem/PythonTests/assetpipeline/scene_settings_tests/tests/scene_settings_manifest_vector_widget_tests_in_editor.py:32]: what is the value of storing the message in a different place? This seems like an unneeded abstraction

Comment (AMZN-stankowi) [reply AutomatedTesting/Gem/PythonTests/assetpipeline/scene_settings_tests/tests/scene_settings_manifest_vector_widget_tests_in_editor.py:32]: I was following the pattern used elsewhere with the Python automated tests. I'm not sure why it's done that way, but it seemed useful to be able to share the messages across multiple tests.

**Review by jackalbe — APPROVED** [review]

**Comment by lmbr-pip** [comment]
Delivered as part of https://github.com/o3de/o3de/issues/12369

**Comment by rainbj** [comment]
@AMZN-stankowi - Thanks for working on this. I'm wondering if this issue with the file name being too long and cutting off the file dirty flag is an issue that already lives elsewhere in the editor. This doesn't seem to be an isolated situation. This makes me wonder if globally we should be prepending the dirty flag to the file name instead of appending it to the end. Example

- (OLD):  OpenFileInEditor.prefab*
- (NEW):  *OpenFileInEditor.prefab

@yuyihsu and @hultonha for visibility

**Pull request merged**